src/components/src/main/java/org/apache/jmeter/assertions/CompareAssertion.java

```diff
@@ -65,78 +65,82 @@ public AssertionResult getResult(SampleResult response) {
     }
 
     private void compareTime(CompareAssertionResult result) {
-        if (compareTime >= 0) {
-            long prevTime = -1;
-            SampleResult prevResult = null;
-            boolean success = true;
-            StringBuilder buf = new StringBuilder();
-            for(SampleResult sResult : responses) {
-                long currentTime = sResult.getTime();
-                if (prevTime != -1) {
-                    success = Math.abs(prevTime - currentTime) <= compareTime;
-                    prevResult = sResult;
-                }
-                if (!success) {
-                    result.setFailure(true);
-                    buf.setLength(0);
-                    appendResultDetails(buf, prevResult);
-                    buf.append(JMeterUtils.getResString("comparison_response_time")).append(prevTime); //$NON-NLS-1$
-                    result.addToBaseResult(buf.toString());
-                    buf.setLength(0);
-                    appendResultDetails(buf, sResult);
-                    buf.append(JMeterUtils.getResString("comparison_response_time")).append(currentTime); //$NON-NLS-1$
-                    result.addToSecondaryResult(buf.toString());
-                   result.setFailureMessage(
-                           JMeterUtils.getResString("comparison_differ_time")+ //$NON-NLS-1$
-                           compareTime+
-                           JMeterUtils.getResString("comparison_unit")); //$NON-NLS-1$
-                    break;
-                }
+        if (compareTime < 0) {
+            return;
+        }
+        long prevTime = -1;
+        SampleResult prevResult = null;
+        boolean success = true;
+        StringBuilder buf = new StringBuilder();
+        for (SampleResult sResult : responses) {
+            long currentTime = sResult.getTime();
+            if (prevTime != -1) {
+                success = Math.abs(prevTime - currentTime) <= compareTime;
+                prevResult = sResult;
+            }
+            if (success) {
                 prevResult = sResult;
                 prevTime = currentTime;
+            } else {
```

Comment:
Ah I thought you would just put `continue` and avoid extra indentation level

Reply:
Oh I see, however, I think this is slightly clearer, as success is most likely so should be first then deal with failure afterwards and use break?

Reply:
I think it should be `if (whatever) { shorter branch; } else { longer branch; }`

Of course, if `shorter branch` ends with `return` or `break` or `continue`, then `else` is not required, and helps to minimize indentation level for `longer branch`

Of course that might depend on case-by-case basis, however most of the times it helps to put shorter fragments first, so the nesting is easier to identify.

On top of that, negated conditions are typically harder to understand, so the following would be so much better than the current code in `master`:

```
if (success) {
   prevResult = sResult;
   prevTime = currentTime;
   continue;
}
// Set failure message and stop the search

...
break;
```

Reply:
So, we do have the shortest (and most likely) branch at the top now? The only difference from your suggestion is I use:
```
...
else {
    ...
    break;
}
```

instead of 
```
if (success) {
    ...
    continue;
}
...
break;
```

which I think it slightly more intention revealing `if else` rather than using `continue` which is just that bit harder to reason about?

Reply:
```
....
...
break;
```

is complicated in any case, so adding yet another nesting layer does not really help there.

Moving some of the code to a smaller method would clarify the intention much better than adding `else`.

You do touch that code, that is why I raise the concerns.
100500 levels of `else` branches never help readability.

Reply:
I wonder, if it would help to change the code from checking for success to checking for failure. That way, we would get a naturally looking `if (failure) { markFailure(...); return }` instead of `break` or `continue`.